src/Illuminate/Session/Middleware/StartSession.php

```diff
@@ -178,9 +177,8 @@ protected function addCookieToResponse(Response $response, SessionInterface $ses
         }
 
         if ($this->sessionIsPersistent($config = $this->manager->getSessionConfig())) {
-            $response->headers->setCookie(new Cookie(
-                $session->getName(), $session->getId(), $this->getCookieExpirationDate(),
-                $config['path'], $config['domain'], Arr::get($config, 'secure', false)
+            $response->headers->setCookie(cookie()->make(
```

Comment:
Wouldn't depending on `cookie()` make it requires the whole framework? 

https://github.com/laravel/framework/blob/5.2/src/Illuminate/Foundation/helpers.php#L231